tcms/static/js/testrun_actions.js

```diff
@@ -311,39 +298,16 @@ var updateCaseRunStatus = function(e) {
     } else {
       fireEvent(link, 'click');
     }
-  };
-
-  // Add comment
-  if (parameters['comment'] != '') {
-    // Reset the content to loading
-    var ajax_loading = getAjaxLoading();
-    ajax_loading.id = 'id_loading_' + parameters['case_id'];
-    container.html(ajax_loading);
-    var c = jQ('<div>');
-
-    if (parameters['value'] != '') {
-        jsonRPC('TestExecution.add_comment', [object_pk, parameters['comment']], function(data){
-            updateCommentsCount(parameters['case_id'], true);
-        });
-    } else {
-        jsonRPC('TestExecution.add_comment', [object_pk, parameters['comment']], function(data){
-            updateCommentsCount(parameters['case_id'], true);
-            callback();
-        });
+  });
+};
 
-    }
-  }
+function addComment(executionId, caseId) {
+  const comment = $('td > textarea[name=comment]').val()
```

Comment:
pass the comment string as argument from the calling context instead of searching on the entire page which is bound for failures and bugs.